crates/iota-core/src/execution_cache.rs

```diff
@@ -913,15 +900,10 @@ macro_rules! implement_passthrough_traits {
                 &'a self,
                 _: &'a EpochStartConfiguration,
             ) -> BoxFuture<'a, ()> {
-                // If we call this method instead of ProxyCache::reconfigure_cache, it's a bug.
-                // Such a bug would almost certainly cause other test failures before reaching
-                // this point, but if it somehow slipped through it is better to crash
-                // than risk forking because ProxyCache::reconfigure_cache was not
-                // called.
-                panic!(
-                    "reconfigure_cache should not be called on a {}",
-                    stringify!($implementor)
-                );
+                // Since we now use WritebackCache directly at startup (if the epoch flag is
+                // set), this can be called at reconfiguration time. It is a no-op.
+                // TODO: remove this once we completely remove ProxyCache.
+                std::future::ready(()).boxed()
```

Comment:
Can we somehow check if the implementor is the proxy cache and then panic with the old message, otherwise do the no-op?

Reply:
Yeah but's the whole point of this panic here, if for whatever reason that slipped somehow through for proxy_cache, it needs to panic here, because then the reconfigure_cache was not called.

Reply:
This is the `reconfigure_cache` of proxy cache, it will not go through the one above
https://github.com/iotaledger/iota/blob/fc692a2807ea994e1e0a0badaa6786b5428026be/crates/iota-core/src/execution_cache/proxy_cache.rs#L384-L389

https://github.com/iotaledger/iota/blob/fc692a2807ea994e1e0a0badaa6786b5428026be/crates/iota-core/src/execution_cache/proxy_cache.rs#L77-L97

Reply:
I don't see how it'll slip through with proxy cache. It only panics if we call `reconfigure_cache` from the Writeback or Passthrough cache in proxy cache directly, but there's no such behaviour in the current codebase.